cmd/ctr/list.go

```diff
@@ -33,69 +30,59 @@ var listCommand = cli.Command{
 		)
 		defer cancel()
 
-		tasks, err := getTasksService(context)
+		client, err := newClient(context)
 		if err != nil {
 			return err
 		}
-
-		containers, err := getContainersService(context)
-		if err != nil {
-			return err
-		}
-
-		response, err := containers.List(gocontext.Background(), &containersapi.ListContainersRequest{})
+		containers, err := client.Containers(ctx)
 		if err != nil {
 			return err
 		}
-
 		if quiet {
-			for _, c := range response.Containers {
-				fmt.Println(c.ID)
+			for _, c := range containers {
+				fmt.Println(c.ID())
 			}
-		} else {
+			return nil
+		}
 
-			tasksResponse, err := tasks.List(ctx, &execution.ListRequest{})
+		w := tabwriter.NewWriter(os.Stdout, 10, 1, 3, ' ', 0)
+		fmt.Fprintln(w, "ID\tIMAGE\tPID\tSTATUS")
+		for _, c := range containers {
+			image, err := c.Image(ctx)
```

Comment:
Running `ctr list` on a system where I have created non-Image based containers (via https://github.com/docker/swarmkit/pull/1965, so not via ctr) results in:
```
$ ./bin/ctr ls 
ctr: container is not based on an image
```
I suppose this was not intended/desired and I should file an issue?